core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -534,6 +534,20 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>, Kokkos::Cuda> {
   const Policy m_rp;
 
  public:
+  template <typename Policy, typename Functor>
+  static int max_tile_size_product(const Policy& pol, const Functor&) {
+    cudaFuncAttributes attr =
+        CudaParallelLaunch<ParallelFor,
+                           LaunchBounds>::get_cuda_func_attributes();
+    auto const& prop = pol.space().cuda_device_prop();
+    // Limits due do registers/SM
```

Comment:
Why are we only considering registers?

Reply:
I stole this code from @crtrott

Reply:
because there is no user side shared memory request for mdrange i.e. it can't be a limit.

Reply:
Resolved via a comment in 5ad44e8